107/src/main/java/org/ehcache/jsr107/EhcacheExpiryWrapper.java

```diff
@@ -35,12 +36,12 @@ public Duration getExpiryForCreation(K key, V value) {
   }
 
   @Override
-  public Duration getExpiryForAccess(K key, V value) {
-    return wrappedExpiry.getExpiryForAccess(key, value);
+  public Duration getExpiryForAccess(K key, Invoker<V> value) {
+    return wrappedExpiry.getExpiryForAccess(key, (Invoker) value);
```

Comment:
Alternatively, you could change the type of the `wrappedExpiry` field to `Expiry<K, V>` and cast the `wrappedExpiry` on the way in using an unchecked cast -- what you're concerned with in this case is solely that the `Expiry` implementation _accept_ `<K,V>` which you know because one handling _super_ was passed in.  This will eliminate the need to cast each value on its way in.